src/pkg/packager/create_stages.go

```diff
@@ -0,0 +1,762 @@
+		}
+
+		if !p.cfg.CreateOpts.SkipSBOM {
+			componentSBOM, err := p.getFilesToSBOM(component)
+			if err != nil {
+				return fmt.Errorf("unable to create component SBOM: %w", err)
+			}
+			if componentSBOM != nil && len(componentSBOM.Files) > 0 {
+				componentSBOMs[component.Name] = componentSBOM
+			}
+		}
+
+		// Combine all component images into a single entry for efficient layer reuse.
+		for _, src := range component.Images {
+			refInfo, err := transform.ParseImageRef(src)
+			if err != nil {
+				return fmt.Errorf("failed to create ref for image %s: %w", src, err)
+			}
+			imageList = append(imageList, refInfo)
+		}
+	}
+
+	imageList = helpers.Unique(imageList)
+	var sbomImageList []transform.Image
+
+	// Images are handled separately from other component assets.
+	if len(imageList) > 0 {
+		message.HeaderInfof("📦 PACKAGE IMAGES")
+
+		p.layout = p.layout.AddImages()
+
+		var pulled []images.ImgInfo
+		var err error
+
+		doPull := func() error {
+			imgConfig := images.ImageConfig{
+				ImagesPath:        p.layout.Images.Base,
+				ImageList:         imageList,
+				Insecure:          config.CommonOptions.Insecure,
+				Architectures:     []string{p.cfg.Pkg.Metadata.Architecture, p.cfg.Pkg.Build.Architecture},
+				RegistryOverrides: p.cfg.CreateOpts.RegistryOverrides,
+			}
+
+			pulled, err = imgConfig.PullAll()
+			return err
+		}
+
+		if err := helpers.Retry(doPull, 3, 5*time.Second, message.Warnf); err != nil {
+			return fmt.Errorf("unable to pull images after 3 attempts: %w", err)
+		}
+
+		for _, imgInfo := range pulled {
+			if err := p.layout.Images.AddV1Image(imgInfo.Img); err != nil {
+				return err
+			}
+			if imgInfo.HasImageLayers {
+				sbomImageList = append(sbomImageList, imgInfo.RefInfo)
+			}
+		}
+	}
+
+	// Ignore SBOM creation if the flag is set.
+	if p.cfg.CreateOpts.SkipSBOM {
+		message.Debug("Skipping image SBOM processing per --skip-sbom flag")
+	} else {
+		p.layout = p.layout.AddSBOMs()
+		if err := sbom.Catalog(componentSBOMs, sbomImageList, p.layout); err != nil {
+			return fmt.Errorf("unable to create an SBOM catalog for the package: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (p *Packager) assembleSkeleton() error {
+	if err := p.skeletonizeExtensions(); err != nil {
+		return err
+	}
+	for _, warning := range p.warnings {
+		message.Warn(warning)
+	}
+	for idx, component := range p.cfg.Pkg.Components {
+		isSkeleton := true
+		if err := p.addComponent(idx, component, isSkeleton); err != nil {
+			return err
+		}
+
+		if err := p.layout.Components.Archive(component, false); err != nil {
+			return err
+		}
+	}
+	checksumChecksum, err := p.generatePackageChecksums()
+	if err != nil {
+		return fmt.Errorf("unable to generate checksums for skeleton package: %w", err)
+	}
+	p.cfg.Pkg.Metadata.AggregateChecksum = checksumChecksum
+
+	return p.writeYaml()
+}
+
+func (p *Packager) output() error {
+	// output assumes it is running from cwd, not the build directory
```

Comment:
This note should probably be above the function so it shows up in p.output intellisense